docs/Deployment.md

```diff
@@ -0,0 +1,99 @@
+# Deploying MsQuic
+
+MsQuic is used as the basis for several different protocols (HTTP, SMB, etc.), but they all have several things in common when it comes time to deploy them. This document outlines the various things that must be taken into account whenever deploying an MsQuic based solution.
+
+# Configuration
+
+MsQuic supports a number of configuration knobs (or settings). These settings can either be set dynamically (via the `QUIC_SETTINGS` structure) or via persistent storage (e.g. registry on Windows).
+
+| Setting                  | Type     | Name                   | Description                                                                                           | Restart<br>Required |
+|--------------------------|----------|------------------------|-------------------------------------------------------------------------------------------------------|---------------------|
+| Max Partition Count      | uint16_t | MaxPartitionCount      | The maximum processor count used for partitioning work in MsQuic                                      | Yes                 |
+| Max Operations per Drain | uint8_t  | MaxOperationsPerDrain  | The maximum number of operations to drain per connection quantum                                      | No                  |
+| Retry Memory Limit       | uint16_t | RetryMemoryFraction    | The percentage of available memory usable for handshake connections before stateless retry is used    | No                  |
+| Max Worker Queue Delay   | uint32_t | MaxWorkerQueueDelayMs  | The maximum queue delay (in ms) allowed for a worker thread                                           | No                  |
+| Max Stateless Operations | uint32_t | MaxStatelessOperations | The maximum number of stateless operations that may be queued at any one time                         | No                  |
+| Initial Window Size      | uint32_t | InitialWindowPackets   | The size (in packets) of the initial congestion window for a connection                               | No                  |
+|                          |          |                        |                                                                                                       |                    |
+
+**TODO** - Finish list above
+
+## Windows
+
+On Windows, these settings can set via the registry. The main registry path for the keys is:
+
```

Comment:
please mention what happens on build to build upgrades. also mention that these only impact new connections?